source/extensions/common/wasm/wasm.h

```diff
@@ -148,16 +172,20 @@ class Context : public Http::StreamFilter,
   // HTTP Filter Callbacks
   //
   // StreamInfo
-  virtual std::string getRequestStreamInfoProtocol();
-  virtual std::string getResponseStreamInfoProtocol();
-  // Metadata: the values are serialized ProtobufWkt::Struct
-  virtual std::string getRequestMetadata(absl::string_view key);
-  virtual void setRequestMetadata(absl::string_view key, absl::string_view serialized_proto_struct);
-  virtual PairsWithStringValues getRequestMetadataPairs();
-  virtual std::string getResponseMetadata(absl::string_view key);
-  virtual void setResponseMetadata(absl::string_view key,
-                                   absl::string_view serialized_proto_struct);
-  virtual PairsWithStringValues getResponseMetadataPairs();
+  virtual std::string getProtocol(StreamType type);
+
+  // Metadata
+  // When used with MetadataType::Request/Response refers to metadata with name "envoy.wasm": the
```

Comment:
we will have multiple filters of type `envoy.wasm` inserted. It is not clear how we can assume just 1.

Reply:
example
```yaml
            name: envoy.wasm
            config:
               Name: istio.metadataFilter
               type: “inbound|outbound|gateway”
               vm_config:
                vm: "envoy.wasm.vm.wavm"
                code:
                  filename: "/var/lib/istio_metadata_http.wasm"
                allow_precompiled: true
```

Reply:
I just checked  experimentally in this branch. Multiple filters of type envoy.wasm, works.